src/libpriv/rpmostree-origin.cxx

```diff
@@ -533,6 +533,27 @@ rpmostree_origin_set_rebase (RpmOstreeOrigin *origin,
   return rpmostree_origin_set_rebase_custom (origin, new_refspec, NULL, NULL, error);
 }
 
+// We tried to use `baserefspec` when changing the origin
+// to something ostree didn't understand, but never completed
+// that work.  This logic just changes `baserefspec` when
+// there are layered packages.
+static void
+sync_baserefspec (RpmOstreeOrigin *origin)
+{
+  if (g_key_file_has_key (origin->kf, "packages", "requested", NULL) || g_key_file_has_key (origin->kf, "packages", "requested-local", NULL))
```

Comment:
Hmm, I think here we could just use `rpmostree_origin_may_require_local_assembly` instead. That'll fix the previous comment and a bunch of other things.

Reply:
Really good review feedback - thanks, fixed!